client/X11/xf_keyboard.c

```diff
@@ -81,14 +84,13 @@ int xf_keyboard_action_script_init(xfContext* xfc)
 	{
 		strtok(buffer, "\n");
 		keyCombination = _strdup(buffer);
-		ArrayList_Add(xfc->keyCombinations, keyCombination);
+		if (ArrayList_Add(xfc->keyCombinations, keyCombination) < 0)
+			return 0;
 	}
 
 	exitCode = pclose(keyScript);
 
-	xf_event_action_script_init(xfc);
-
-	return 1;
+	return xf_event_action_script_init(xfc);
```

Comment:
This function (xf_keyboard_action_script_init) seemed to return 0 on failure and 1 on success. Here you return the result of xf_event_action_script_init() which returns -1 on failure